pyfatfs/PyFat.py

```diff
@@ -1052,11 +1053,13 @@ def mkfs(self, filename: str,
         self.__set_fp(open(filename, mode='rb+'))
 
         if size is None:
-            size = self.__fp_offset - self.__fp.seek(-1)
+            self.__fp.seek(0, 2)
+            size = self.__fp.tell()
```

Comment:
`seek()` returns the position it seeked to, no need for `tell()`

```python
>>> from io import open
>>> foo = open("/dev/sda", "rb+")
>>> foo.seek(0, 2)
16000900661248
>>> foo.tell()
16000900661248
```

Reply:
Also: This will initialize a wrong size, as it ignores the offset. If an offset is specified it should be subtracted when determining the size. It will also break the truncate call in line 1061 as it will try to overprovision the file (which fails on fixed size / block devices).